src/main/java/com/abdullah/coding/challenge/services/BookingService.java

```diff
@@ -0,0 +1,359 @@
+package com.abdullah.coding.challenge.services;
+
+import com.abdullah.coding.challenge.Response;
+import com.abdullah.coding.challenge.dto.BookingDTO;
+import com.abdullah.coding.challenge.entities.Booking;
+import com.abdullah.coding.challenge.entities.Cab;
+import com.abdullah.coding.challenge.entities.Customer;
+import com.abdullah.coding.challenge.enums.BookingStatus;
+import com.abdullah.coding.challenge.enums.CustomerStatus;
+import com.abdullah.coding.challenge.repository.BookingRepository;
+import com.abdullah.coding.challenge.util.TimeUtil;
+import org.springframework.beans.factory.annotation.Autowired;
+import org.springframework.http.HttpStatus;
+import org.springframework.stereotype.Service;
+
+import javax.transaction.Transactional;
+import java.sql.Timestamp;
+import java.util.List;
+import java.util.Optional;
+import java.util.UUID;
+
+@Service
+public class BookingService {
+    @Autowired
+    private BookingRepository repository;
```

Comment:
the name has to be bookingRepository so that the code can have a good visibility

Reply:
You could but if you notice one thing in services is that they only know about their related repository. They do not know anything about other repositories. So, by just naming, "repository", it denotes that this particular service only knows and cares about this particular repository. You could name it as bookingRespository but it's not really 'needed'.